src/runtime/internal/scheduler.ts

```diff
@@ -43,14 +45,19 @@ export function flush() {
 		for (let i = 0; i < dirty_components.length; i += 1) {
 			const component = dirty_components[i];
 			set_current_component(component);
-			update(component.$$);
+			const is_dirty_from_binding = dirty_binding_component_map.has(component.constructor.name);
+			update(component.$$, is_dirty_from_binding);
```

Comment:
IIUC `dirty_binding_component_map` can also be named as `previous_dirty_components`. And that we check if a component is re-marked dirty, which if so, `update()` will not call `before_update`.

I think we can refactor the names and prevent using a Map for `previous_dirty_components`. Copying over like `previous_dirty_components = [...dirty_components]`, and comparing with `previous_dirty_components.includes(component)` should work.

I'm not sure if this it *the* way of solving the issue though, someone else may have a better overview of this, but here are some things I found which can make the code a bit easier to follow.

Reply:
Thank you for the review I will update the code soon 👍 
`someone else may have a better overview of this` that will be great 👍